lamp.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python3
+
+from __future__ import print_function
+
+import argparse
+import logging
+import os
+import platform
+import shutil
+import subprocess
+import sys
+import tarfile
+import urllib.request
+
+
+def get_toolchain_url(os_family, distro=None):
+    if os_family == 'Darwin':
+        build_id = 'genny_toolchain_macos_1014_b15f51760a9e651fb66bc0593e2eea738e15ffa2_19_02_07_07_04_06'
+    elif os_family == 'Linux':
+        if distro == 'ubuntu1804':
+            build_id = 'genny_toolchain_ubuntu1804_b15f51760a9e651fb66bc0593e2eea738e15ffa2_19_02_07_07_04_06'
+        elif distro == 'rhel7':
+            build_id = 'genny_toolchain_rhel70_b15f51760a9e651fb66bc0593e2eea738e15ffa2_19_02_07_07_04_06'
+        elif distro == 'amazon2':
+            build_id = 'genny_toolchain_amazon2_b15f51760a9e651fb66bc0593e2eea738e15ffa2_19_02_07_07_04_06'
+        elif distro == 'arch':
+            build_id = 'genny_toolchain_archlinux_b15f51760a9e651fb66bc0593e2eea738e15ffa2_19_02_07_07_04_06'
+        else:
+            raise ValueError('Unknown Linux distro {}; please set "--linux-distro"'.format(distro))
+    else:
+        raise ValueError('Unsupported platform: {}'.format(os_family))
+
+    url = 'https://s3.amazonaws.com/mciuploads/genny-toolchain/{}/gennytoolchain.tgz'.format(
+        build_id)
+    logging.debug('gennytoolchain URL: %s', url)
+    return url
+
+
+def fetch_and_install_toolchain(url, install_dir):
+    if not os.path.exists(install_dir):
+        logging.critical(
+            'Please create the parent directory for gennytoolchain: '
+            '`sudo mkdir -p %s; sudo chown $USER %s`', install_dir, install_dir)
+        sys.exit(1)
+
+    if not os.access(install_dir, os.W_OK):
+        logging.critical(
+            'Please ensure you have write access to the parent directory for gennytoolchain: '
+            '`sudo chown $USER %s`', install_dir)
+        sys.exit(1)
+
+    toolchain_tarball = os.path.join(install_dir, 'gennytoolchain.tgz')
+    if os.path.isfile(toolchain_tarball):
+        logging.info('Skipping downloading %s', toolchain_tarball)
+    else:
+        logging.info('Downloading gennytoolchain, please wait...')
+        urllib.request.urlretrieve(url, toolchain_tarball)
+        logging.info('Finished Downloading gennytoolchain as %s', toolchain_tarball)
+
+    toolchain_dir = os.path.join(install_dir, 'gennytoolchain')
+    if os.path.exists(toolchain_dir):
+        logging.info('Skipping extracting the toolchain into: %s', toolchain_dir)
+    else:
+        logging.info('Extracting gennytoolchain, please wait...')
+        tarball = tarfile.open(toolchain_tarball)
+        tarball.extractall(toolchain_dir)
+        logging.info('Finished extracting gennytoolchain into %s', toolchain_dir)
```

Comment:
Yeah I understand your sentiment, which is why I didn't include the URL and the output dir. The otherwise useless log here is just to tell the user the script is working. Without it, it looks like the script hangs with no output at all.

The logging.info below are to prepare users for the cmake/ninja/ctest output. I want to log the command we used so the cmake output can make sense if people.